.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,58 @@
+{
+	"name": "Tensorflow Dev Container",
+	
+	"dockerFile": "Dockerfile",
+	
+	// NOTE: This is currently not supported unitll we
+	//       separate CPU and GPU layers
+	//       https://github.com/tensorflow/build/pull/47
+	// Uncomment this for GPU image
+	// "build": { 
+	// 		"args": { 
+	//			"IMAGE_TYPE": "-gpu"
+	//		} 
+	//	},
+	
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": {
+		"terminal.integrated.shell.linux": "/bin/bash",
+		"python.testing.pytestEnabled": true,
+		"python.testing.pytestArgs": [ 
+			"./tensorflow"
+		],
+		"C_Cpp.clang_format_style": "{BasedOnStyle: Google}",
+		"C_Cpp.default.includePath": [
+			"${workspaceFolder}/**",
+        	],
+		"files.watcherExclude": {
+			"**/bazel-*/**": true
+		},
+		"search.exclude": {
+			"**/bazel-*/**": true
+		}
+	},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"ms-python.python",
+		"ms-vscode.cpptools",
+	],
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	// "forwardPorts": [],
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	// "postCreateCommand": "uname -a",
+    "mounts": [
+        "source=tensorflow_cache,target=/root/.cache/bazel/"
```

Comment:
@chuxel Do you know why also using this the `bazel` cache is invalidate on every vscode restart?

Reply:
I've debugged a little more and it is seem to related `conde insider` (dality) update restart. Let me know if you want an upstream ticket.

Reply:
@Chuxel  I've an upstream ticket now at https://github.com/microsoft/vscode-dev-containers/issues/1453